bson/src/main/org/bson/Vector.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson;
+
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import static org.bson.assertions.Assertions.assertNotNull;
+import static org.bson.assertions.Assertions.isTrue;
+import static org.bson.assertions.Assertions.isTrueArgument;
+import static org.bson.assertions.Assertions.notNull;
+
+/**
+ * Represents a vector that is stored and retrieved using the BSON Binary Subtype 9 format.
+ * This class supports multiple vector {@link Dtype}'s and provides static methods to create
+ * vectors and methods to retrieve their underlying data.
+ * <p>
+ * Vectors are densely packed arrays of numbers, all the same type, which are stored efficiently
+ * in BSON using a binary format.
+ *
+ * @mongodb.server.release 6.0
+ * @see BsonBinary
+ * @since BINARY_VECTOR
+ */
+public final class Vector {
```

Comment:
Let's bikeshed on the name and consider more description options like

* `BinaryVector`
* `BsonBinaryVector`

This corresponds a bit better to the ` BsonValue` type hierarchy, which has the `Bson` prefix for all the types.

Reply:
I understand the idea of aligning the name with the `BsonValue` hierarchy, however since` Vector` isn't a direct part of that hierarchy (similar to how UUID isn't), I believe it would be more consistent to avoid the `Bson` prefix.

Regarding `BinaryVector`, the term "binary vector" seems to be commonly used in machine learning to refer to binary quantized vectors, like our `PackedBitVector`, which could lead to confusion. For reference, here’s an example: [Binary Vector in Machine Learning](https://machinelearningcatalogue.com/input_data_type/idt_binary-vector.html#:~:text=A%20binary%20vector%20is%20a,vectors%20and%20vectors%20of%20variables).

I personally think `Vector` works, but there is the potential collision with the `Vector collection class`, and IDEs often auto-import it, which could be problematic. I think we should brainstorm further on the naming. Perhaps `NumericVector` could be an option, but it’s worth giving it more thought.

Reply:
If it wasn't for `java.util.Vector` I'd probably let this go.  But let's get a second opinion before resolving.

Reply:
@stIncMale I saw your thumbs up but not sure what it means. Let it go or continue to bikeshed?

Reply:
I haven't thumbsed up, that's not mine. However, I don't have any suggestions, nor do I object to `Vector`.

P.S. I do object to both Java SE and MongoDB calling vector components "elements", but I can't do anything with people disregarding actual terms and using arbitrary words instead.